cli/code_scanner.py

```diff
@@ -531,3 +553,45 @@ def _does_severity_match_severity_threshold(severity: str, severity_threshold: s
         return True
 
     return detection_severity_value >= Severity.try_get_value(severity_threshold)
+
+
+def _get_scan_result(cycode_client, scan_async_result: ScanInitializationResponse,
+                     scan_details: ScanDetailsResponse) -> ZippedFileScanResult:
+    scan_result = ZippedFileScanResult(did_detect=False, detections_per_file=[],
+                                       scan_id=scan_async_result.scan_id,
+                                       report_url=scan_details.report_url)
+    if not scan_details.detections_count:
+        return scan_result
+
+    wait_for_detections_creation(cycode_client, scan_async_result.scan_id, scan_details.detections_count)
+    scan_detections = cycode_client.get_scan_detections_with_paging(scan_async_result.scan_id)
+    scan_result.detections_per_file = _map_detections_per_file(scan_detections)
+    scan_result.did_detect = True
+    return scan_result
+
+
+def wait_for_detections_creation(cycode_client, scan_id: str, expected_detections_count: int):
```

Comment:
Please make sure that the detection count that you have in the scan is the real detection count.
I think it is the count of violations before the classification rules exclude some violations.

Reply:
good point. will verify.

Reply:
Verified?